modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java

```diff
@@ -788,4 +836,269 @@ else if (e.getCause() instanceof IgniteException)
         else
             return new IgniteException(e.getCause());
     }
+
+    /**
+     * Asynchronous calculation of caches size with divided by tables.
+     *
+     * @param grpCtx Cache group context.
+     * @param locPart Local partition.
+     * @return Future with cache sizes.
+     */
+    private Future<CacheSize> calcCacheSizeAsync(
+        CacheGroupContext grpCtx,
+        GridDhtLocalPartition locPart
+    ) {
+        return calcExecutor.submit(() -> {
+            try {
+                int grpId = grpCtx.groupId();
+
+                if (failCalcCacheSizeGrpIds.contains(grpId))
+                    return new CacheSize(null, null);
+
+                boolean reserve = false;
+
+                int partId = locPart.id();
+
+                try {
+                    if (!(reserve = locPart.reserve()))
```

Comment:
Why do not rewrite it different?
reserve != locPart.reserve()

Reply:
I think we should leave it as it is.